src/components/QueryEditor.tsx

```diff
@@ -111,61 +45,66 @@ export function QueryEditor(props: Props) {
-    const next = { ...props.query, format: e.value !== undefined ? e.value : QueryFormat.Table };
-    props.onChange(next);
-    processQuery(next);
-  };
-
-  const onLocationChange = (e: SelectableValue) => {
-    const next = { ...props.query, location: e.value || DEFAULT_REGION };
-    props.onChange(next);
-    processQuery(next);
+  const onColumnsChange = (q: BigQueryQueryNG) => {
+    onChange(q);
+    processQuery(q);
   };
 
-  const onRawQueryChange = useCallback(
-    (q: BigQueryQueryNG) => {
-      onChange(q);
-      processQuery(q);
-    },
-    [onChange, processQuery]
-  );
-
   if (apiLoading || apiError || !apiClient) {
     return null;
   }
 
   return (
-    <div>
-      <HorizontalGroup>
-        <Field label="Processing location">
-          <Select
-            options={PROCESSING_LOCATIONS}
-            value={queryWithDefaults.location}
-            onChange={onLocationChange}
-            className="width-12"
-            menuShouldPortal={true}
-          />
-        </Field>
-
-        <Field label="Format as">
-          <Select
-            options={QUERY_FORMAT_OPTIONS}
-            value={queryWithDefaults.format}
-            onChange={onFormatChange}
-            className="width-12"
-            menuShouldPortal={true}
-          />
-        </Field>
-      </HorizontalGroup>
-
-      <QueryEditorRaw
-        getTables={getTables}
-        getColumns={getColumns}
+    <>
+      <QueryHeader
+        onChange={onChange}
+        onRunQuery={onRunQuery}
+        onQueryRowChange={setQueryRowFilter}
+        queryRowFilter={queryRowFilter}
         query={queryWithDefaults}
-        onChange={onRawQueryChange}
-        onRunQuery={props.onRunQuery}
+        // TODO: add proper dirty check
+        sqlCodeEditorIsDirty={!!queryWithDefaults.sql.columns?.length}
+        apiClient={apiClient}
       />
-    </div>
+
+      <Space v={0.5} />
+
+      {queryWithDefaults.editorMode !== EditorMode.Code && (
+        <EditorRows>
+          <EditorRow>
+            <SQLBuilderSelectRow query={queryWithDefaults} onQueryChange={onColumnsChange} apiClient={apiClient} />
+          </EditorRow>
+          {queryRowFilter.group && (
+            <EditorRow>
+              <EditorField label="Group by column">
+                <SQLGroupByRow query={queryWithDefaults} onQueryChange={onColumnsChange} apiClient={apiClient} />
+              </EditorField>
+            </EditorRow>
+          )}
+          {queryRowFilter.order && (
+            <EditorRow>
+              <SQLOrderByRow query={queryWithDefaults} onQueryChange={onColumnsChange} apiClient={apiClient} />
+            </EditorRow>
+          )}
+          {queryRowFilter.preview && queryWithDefaults.rawSql && (
+            <EditorRow>
+              <EditorField label="Preview">
+                {/* TODO: Figure out how to not hardcode width */}
+                <CodeEditor height={80} width={800} language="sql" value={queryWithDefaults.rawSql} readOnly={true} />
```

Comment:
This needs to be wrapped to an additional container that has a sensor hook attached that calculates the available width unfortunately.